cranelift-wasm/src/code_translator.rs

```diff
@@ -1741,3 +1753,29 @@ fn pop2_with_bitcast(
     let bitcast_b = optionally_bitcast_vector(b, needed_type, builder);
     (bitcast_a, bitcast_b)
 }
+
+/// A helper for bitcasting a sequence of arguments; to avoid SIMD type issues, this will bitcast
+/// vectors to the type expected in their next use.
+fn bitcast_arguments(
+    args: &[Value],
+    params: &[AbiParam],
+    builder: &mut FunctionBuilder,
+) -> Vec<Value> {
```

Comment:
Would it be possible to make this function take `args` by `&mut [Value]` and mutate it in place, instead of returning a new `Vec`? At the places where this is called, it's passing the result from `peekn`, so it'd require creating a `peekn_mut` or so. Assuming that works, that would avoid a dynamic allocation for each `End`.